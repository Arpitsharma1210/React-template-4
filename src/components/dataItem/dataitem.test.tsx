import React from 'react';
import { render, screen } from '@testing-library/react';
import DataItem from './';

describe('DataItem Component', () => {
  it('renders data item with provided data', () => {
    render(<DataItem type="activejobs" data="100" />);
    expect(screen.getByAltText('dataitem icon')).toBeInTheDocument();
    expect(screen.getByText('Active Jobs')).toBeInTheDocument();
    expect(screen.getByText('100')).toBeInTheDocument();
  });

  it('renders data item with stattype and statdata', () => {
    render(<DataItem type="applicants" data="200" stattype="increase" statdata="5" />);
    expect(screen.getByAltText('dataitem icon')).toBeInTheDocument();
    expect(screen.getByText('Applicants')).toBeInTheDocument();
    expect(screen.getByText('200')).toBeInTheDocument();
    expect(screen.getByAltText('')).toBeInTheDocument(); // Assuming the alt text is empty for the stat icon
    expect(screen.getByText('5%')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<DataItem type="activejobs" data="100" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
