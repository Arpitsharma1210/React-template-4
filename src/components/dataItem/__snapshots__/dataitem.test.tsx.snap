// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataItem Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnyFh dLJTfV"
  >
    <div
      class="imgContainer"
    >
      <img
        alt="dataitem icon"
        src="test-file-stub"
      />
    </div>
    <div
      class="databox"
    >
      <div
        class="title"
      >
        Active Jobs
      </div>
      <div
        class="data"
      >
        100
      </div>
    </div>
  </div>
</DocumentFragment>
`;
